### PR TITLE
Fix MixupAndCutmix float16 support and also allow for one-hot encoded labels

### PR DESCRIPTION
**Pull request by usryokousha**
# Description
Added support to MixupAndCutmix for both sparse and one-hot encoded input labels.  Also ensured that float16 image inputs are cast as float32 for mixup and cutmix ops while preserving the input data type on output from distort method.

> :memo: Please include a summary of the change. 
>  
> * Please also include relevant motivation and context.  
> * List any dependencies that are required for this change.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [ ] TensorFlow 2 migration
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] A new research paper code implementation
- [ ] Other (Specify)

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**: 
I added an additional test to the augment_test.py file testing float16 image inputs and one-hot encoded label inputs.
I successfully passed this new test and all other existing tests ensuring that previous functionality is preserved.

## Checklist

- [x ] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [ x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x ] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ x] I have commented my code, particularly in hard-to-understand areas.
- [ x] I have made corresponding changes to the documentation.
- [ x] My changes generate no new warnings.
- [ x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/vision/beta/ops/augment_test.py [hunk 1]
```diff
@@ -28,308 +28,328 @@
 
 
 def get_dtype_test_cases():
-  return [
-      ('uint8', tf.uint8),
-      ('int32', tf.int32),
-      ('float16', tf.float16),
-      ('float32', tf.float32),
-  ]
+    return [
```
Comment (arashwan) [inline official/vision/beta/ops/augment_test.py:31]: can you revert this unrelated changes?

File: official/vision/beta/ops/augment.py [hunk 2]
```diff
@@ -33,280 +33,279 @@
 
 from tensorflow.python.keras.layers.preprocessing import image_preprocessing as image_ops
```
Comment (arashwan) [inline official/vision/beta/ops/augment.py:34]: For one_hot, can you just add a condition here (https://github.com/tensorflow/models/blob/master/official/vision/beta/tasks/image_classification.py#L230) to not convert to one_hot if cutmix is on?

**Comment by saberkun** [comment]
The code is stale and moved. I think we should close this PR.

**Comment by fyangf** [comment]
No response from the PR author so closing it.
